v3/src/components/web-view/request-queue.ts

```diff
@@ -0,0 +1,30 @@
+import { action, makeObservable, observable } from "mobx"
+import { DIRequest, DIRequestCallback } from "../../data-interactive/data-interactive-types"
+
+type RequestPair = { request: DIRequest, callback: DIRequestCallback }
+export class RequestQueue {
+  @observable
```

Comment:
```suggestion
  @observable.shallow
```
We don't want/need to observe the contents of the items in the queue.